c10/metal/special_math.h

```diff
@@ -311,5 +311,74 @@ float log_gamma(const T x) {
   return LOG_PI - rc - ::metal::log(log_arg);
 }
 
+float zeta(float x, float q) {
+  constexpr float MACHEP = 1.11022302462515654042E-16;
+  const float ZETA_EXPANSION[] = {
```

Comment:
Nit: this array should also be constexpr

Reply:
Done, thanks for the review